### PR TITLE
Implement editor from designs

### LINKED ISSUE #690: Implement editor from designs

### LINKED ISSUE #706: Preview threads & replies
I think it could be useful to have a preview option option before saving a thread or reply. This can prevent some messages that aren't formatted on the prefered way.

An option is the way that GitLab does, you have 2 tabs. The first is where you write and the second is the preview. If you switch to the second you get a call to the server that returns the preview. (In fact the same technique that livewire used.)

### PR DESCRIPTION
**Pull request by joedixon**
This PR implements the new designs for the editor which resolves #690. Additionally, it leverages Livewire to add a new `Preview` feature as suggested by @tvbeek and so resolves #706. This allows users to write their thread, reply or article and preview it before saving.

Screenshot 2021-10-31 at 20 49 22

https://user-images.githubusercontent.com/3438564/139600822-b5287df8-97a6-4a29-b3d9-a479a6bebd64.mov

Screenshot 2021-10-31 at 20 45 27

Screenshot 2021-10-31 at 20 45 08

### CODE REVIEW

File: resources/js/editor.js [hunk 1]
```diff
@@ -61,13 +61,11 @@ window.editorConfig = () => {
                 after: ')',
             },
         },
+        body: body,
+        mode: 'write',
     };
 };
 
-window.expand = (element, minHeight) => {
```
Comment (joedixon) [inline resources/js/editor.js:67]: This is now handled by Alpine and Tailwind.

File: tailwind.config.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 const defaultTheme = require('tailwindcss/defaultTheme');
 
 module.exports = {
+    mode: 'jit',
```
Comment (joedixon) [inline tailwind.config.js:4]: 🔥

File: resources/views/articles/_form.blade.php [hunk 3]
```diff
@@ -20,9 +20,7 @@ class="form-control"
         @formGroup('body')
             <label for="body">Body</label>
 
-            @include('_partials._editor', [
-                'content' => isset($article) ? $article->body() : null
-            ])
+            <x-editor :content="isset($article) ? $article->body() : null"/>
```
[the remark this answers is not in the document]
Comment (joedixon) [reply resources/views/articles/_form.blade.php]: I couldn't use this as the variable wasn't null, but instead, there were certain scenarios where it didn't exist. However, I got around this by extracting a blade component and defaulting to `null`.

**Review by driesvints — CHANGES_REQUESTED** [review]
Let's sneak in https://github.com/laravelio/laravel.io/issues/674 as well.

**Comment by joedixon** [comment]
This also now resolve #674

**Pull request merged**